src/main/java/com/google/sps/app/servlets/SignUp.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.app.servlets;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.gson.Gson;
+import com.google.sps.util.IDgenerator;
+import com.google.sps.util.Validator;
+
+@WebServlet("/SignUp")
+public class SignUp extends HttpServlet{
+    
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String username = request.getParameter("text-input-user-name");
+        
+        if(!Validator.validUserName(username)) {
+            response.getWriter().println("Invalid Username");
```

Comment:
Maybe throw an error with more explicit suggestions? For instance "Username contains invalid characters"

Reply:
I agree! Done.